lib/engine/parser/split_treatment.rb

```diff
@@ -2,20 +2,38 @@ module SplitIoClient
   module Engine
     module Parser
       class SplitTreatment
-        def initialize(segments_repository)
+        def initialize(segments_repository, splits_repository, multiple = false)
+          @splits_repository = splits_repository
           @segments_repository = segments_repository
+          @multiple = multiple
+          @cache = {}
```

Comment:
Since new `@cache` instance gets created at each instantiation of `SplitTreatment`, I pass one `SplitTreatment` instance when `get_treatments` is called, hence cache is invalidated for each `get_treatments` / `get_treatment` call